tv_apps/tv-ftu/js/index.js

```diff
@@ -0,0 +1,129 @@
+/* global SettingsListener, KeyNavigationAdapter, SpatialNavigator */
+(function (exports) {
+'use strict';
+
+window.addEventListener('load', function() {
+
+  var appUsageSlide = {
+
+    init: function () {
+      this.APP_USAGE_ENABLE_KEY = 'metrics.selectedMetrics.level';
+      this.APP_USAGE_ENABLE_VALUE = 'Basic';
+      this.BASE_REM = 10; // 10px
+      this.MAX_SCROLL_VALUE = 1; // 1rem
+
+      this._currentScrollValue = 0; // in rem
+
+      this._noButton =
+          document.getElementById('appUsageSlide-noButton');
+      this._yesButton =
+          document.getElementById('appUsageSlide-yesButton');
+      this._terms =
+          document.getElementById('appUsageSlide-terms');
+
+      this.keyNavAdapter = new KeyNavigationAdapter();
+      this.keyNavAdapter.init();
+      this.keyNavAdapter.on('move', this.onKeyMove.bind(this));
+      this.keyNavAdapter.on('enter-keyup', this.onKeyEnterUP.bind(this));
+
+      this.spatialNav = new SpatialNavigator([
+        this._terms,
+        this._noButton,
+        this._yesButton
+      ]);
+      this.spatialNav.on('focus', this.onFocus.bind(this));
+      this.spatialNav.focus(this._yesButton);
+
+      this._termsSectionRect =
+          document.getElementById('appUsageSlide-termsSection')
+                  .getBoundingClientRect();
+    },
+
+    /**
+     * @return {number} the scrolled value. O means no gap for scrolling
+     */
+    scrollTerms: function (dir) {
+      var gap;
+      var scrollValue = 0;
+      var termsRect = this._terms.getBoundingClientRect();
+      switch (dir) {
+        case 'up':
+          gap =  this._termsSectionRect.top - termsRect.top;
+          // Still have terms overflows the terms section's top area.
+          if (gap > 0) {
+            gap = gap / this.BASE_REM; // Turn px into rem
+            scrollValue = Math.min(gap, this.MAX_SCROLL_VALUE);
+          }
+        break;
+
+        case 'down':
+          gap = termsRect.bottom - this._termsSectionRect.bottom;
+          // Still have terms overflows the terms section's bottom area
+          if (gap > 0) {
+            gap = gap / this.BASE_REM;
+            scrollValue = -1 * Math.min(gap, this.MAX_SCROLL_VALUE);
+          }
+        break;
+      }
+
+      if (scrollValue) {
+        this._currentScrollValue += scrollValue;
+        this._terms.style.transform = 'translateY(' +
+            this._currentScrollValue + 'rem)';
+      }
+      return scrollValue;
```

Comment:
I think adding/subtracting `scrollTop` is just plenty if you are not going to do smooth scrolling... but it's ok for me.